src/pages/teams/[teamId]/dashboard/requests/new/[step].tsx

```diff
@@ -0,0 +1,216 @@
+  EvidenceRequestRequest,
+  EvidenceRequestForm,
+  InternalApiGateway,
+} from '../../../../../../gateways/internal-api';
+import { schemaNewRequestFormStep1 } from '../../../../../../components/NewRequestFormStep1';
+import { schemaNewRequestFormStep2 } from '../../../../../../components/NewRequestFormStep2';
+import { RequestAuthorizer } from '../../../../../../services/request-authorizer';
+import { TeamHelper } from '../../../../../../services/team-helper';
+import { Team } from '../../../../../../domain/team';
+import { User } from '../../../../../../domain/user';
+import * as Yup from 'yup';
+import { Form, Formik } from 'formik';
+import { useRouter } from 'next/router';
+
+type RequestsNewPageProps = {
+  documentTypes: DocumentType[];
+  team: Team;
+  user: User;
+};
+
+const schema = [
+  schemaNewRequestFormStep1,
+  schemaNewRequestFormStep2,
+  Yup.object().shape({}),
+];
+
+const RequestsNewPage: NextPage<WithUser<RequestsNewPageProps>> = ({
+  documentTypes,
+  team,
+  user,
+}) => {
+  const { push, query, replace } = useRouter();
+  const [complete, setComplete] = useState(false);
+  const [submitError, setSubmitError] = useState(false);
+  const [deliveryMethods, setDeliveryMethods] = useState<string[]>(['']);
+  const [previousStepNumber, setPreviousStepNumber] = useState(0);
+
+  let currentStep: number = Array.isArray(query.step)
+    ? parseInt(query.step[0])
+    : parseInt(query.step || '');
+
+  useEffect(() => {
+    window.onpopstate = () => {
+      setPreviousStepNumber(previousStepNumber - 1);
+    };
+    if (currentStep - 1 !== previousStepNumber) {
+      replace(`/teams/${team.id}/dashboard/requests/new/1`);
+    }
+  }, [currentStep]);
+
+  const initialValues: EvidenceRequestForm = {
+    resident: {
+      name: '',
+      email: '',
+      phoneNumber: '',
+    },
+    team: team.name,
+    reason: team.reasons[0].name,
+    documentTypes: [],
+    emailCheckbox: [],
+    phoneNumberCheckbox: [],
+  };
+
+  const steps = [1, 2, 3];
+
+  function renderStepContent(step: number) {
+    switch (step) {
+      case 1:
+        return <NewRequestFormStep1 team={team} />;
+      case 2:
+        return <NewRequestFormStep2 documentTypes={documentTypes} />;
+      case 3:
+        return (
+          <ConfirmRequestDialog
+            documentTypes={documentTypes}
+            deliveryMethods={deliveryMethods}
+            onAccept={submitHandler}
+            onDismiss={() => {
+              push(
+                `/teams/${team.id}/dashboard/requests/new/${currentStep - 1}`
+              );
+              setPreviousStepNumber(previousStepNumber - 1);
+            }}
+          />
+        );
+      default:
+        return <div>Not Found</div>;
+    }
+  }
+
+  const currentValidationSchema = schema[currentStep - 1];
+
+  const submitHandler = async (values: EvidenceRequestForm) => {
+    const payload = buildEvidenceRequestRequest(values);
+    if (currentStep < steps.length) {
+      currentStep += 1;
```

Comment:
Editing this value seems unexpected. Either we should infer it from `router.query.step` or we should have it in state (ideally the former)?

Reply:
addressed